src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.module.Module;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.TaskDescription;
+import seedu.address.model.task.TaskStatus;
+
+/**
+ * Mark a task identified using it's displayed index from the task list as complete.
+ */
+public class MarkCommand extends Command {
+
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Marks the task identified by the index number used in the displayed task list.\n"
+        + "Parameters: INDEX (must be a positive integer)\n"
+        + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_TASK_SUCCESS = "Marked Task: %1$s";
+
+    private final Index targetIndex;
+
+    public MarkCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToMark = lastShownList.get(targetIndex.getZeroBased());
+        Task editedTask = createEditedTask(taskToMark);
+        model.setTask(taskToMark, editedTask);
+        return new CommandResult(String.format(MESSAGE_MARK_TASK_SUCCESS, taskToMark));
+    }
+
+    /**
+     * Creates and returns a {@code Task} with the details of {@code taskToEdit}
+     * marked as complete.
+     */
+    private static Task createEditedTask(Task taskToEdit) {
+        assert taskToEdit != null;
+
+        Module taskModule = taskToEdit.getModule();
+        TaskDescription taskDescription = taskToEdit.getDescription();
+
+        Task editedTask = new Task(taskModule, taskDescription, TaskStatus.COMPLETE);
```

Comment:
It would be better to create a method in task to change the completeness status instead of retrieving module and description and creating a new task

Reply:
You're right, I overlooked that part. I've changed the code accordingly.